### PR TITLE
Add missing game rules, support AI mode, improve the performance.

### PR DESCRIPTION
**Pull request by radi-cho**
Working on `Play vs Computer mode` /AI mode/ - #2 
- [x] Add simple chess algorithm;
- [x] Add [threefold repetition](https://github.com/RSG-Group/Chess/issues/5); (in progress)
- [x] Add the [Fifty move rule](https://en.wikipedia.org/wiki/Fifty-move_rule);
- [x] Add strong AI using minimax algorithm; (~partly~ done)
- [x] Use a/b pruning to improve the minimax;
- ~Improve the evaluation function. [More info](https://chessprogramming.wikispaces.com/Simplified+evaluation+function).~ (planned)

UPDATE:
- [x] Adapt the UI and improve the settings.

### CODE REVIEW

**Comment by retrixe** [comment]
Aha
This should be 2.0 though 🤔

**Comment by radi-cho** [comment]
It'll be 1.1.0, the 2.0 maybe will support multiplayer and few Google Play Games rewards.

**Comment by radi-cho** [comment]
Current issues:
 - After some moves the black figs can't make their move. ~Not~ fixed /testing/.
 - After black pawn go to `game.board[0]` it won't be promoted. Not fixed. In progress.
 - Sometimes when you "deselect" selected cell the black AI will move. ~Not~ fixed.
These things will be fixed up to 2-3 days, followed by the minimax algorithm /the second stage of my plan/.

File: js/game.js [hunk 1]
```diff
@@ -164,4 +196,96 @@ Game.prototype.warning = function (color) {
...
+    turn[turn.length - 1].color === 'W' ? 'b' : 'w';
+ 
+  FEN += ' ' + activeColor;
+ 
+  // Check castling availability
+  var possibleCastling = '';
+ 
+  var whiteKingMoved;
+  var blackKingMoved;
+  turn.some(function (turn) {
```
Comment (samet-karaibryamov) [inline js/game.js]: It's probably better to use `forEach` instead of `some` here.

Comment (radi-cho) [reply js/game.js]: Oh yeah, I'll replace that. Thanks.

**Comment by radi-cho** [comment]
- Added the threefold repetition rule.
- Removed the useless FEN functions (they'll be added into [Chess API](https://github.com/RSG-Group/RSG-Chess-API)),
Check out the discussion in #7 for more information.
Thanks to @samet-karaibryamov for the reviews!
- Disable dismissing the checkmate dialog.

**Comment by radi-cho** [comment]
The threefold repetition and the fifty-move rule are done. Now we can port them to the `Chess mobile` and `Chess desktop` and build the basic `Chess APIs`.

**Comment by radi-cho** [comment]
It'll be a lot easier to loop the valid moves if we have all them in one place. That's why I added this: [Chess-API@`fd1935a`](https://github.com/RSG-Group/RSG-Chess-API/commit/fd1935a7480e5d92905015d57a63cab1dbafcdbd). Soon it may be implemented into the core project...

**Comment by radi-cho** [comment]
I pushed support for minimax algorithm. There are still few issues with it but it's good for now.
Also we need to support timing... For example the user has 10 minutes to make a move or something similar...

**Comment by radi-cho** [comment]
The AI is now strong and stable. We need to adapt the UI and add some features. Then this will be merged and future AI improvements will be pushed to `master` or new branches.

**Comment by radi-cho** [comment]
This is done. If someone find any bug please open an issue to fix it!

**Pull request merged**